### PR TITLE
Core include cleanup

### PR DESCRIPTION
**Pull request by petersteneteg**
Thank you for contributing to the Inviwo repository! Great code makes great apps so please ensure the following:

### CODE REVIEW

File: include/inviwo/core/datastructures/light/directionallight.h [hunk 1]
```diff
@@ -49,7 +47,7 @@ class DirectionalLight : public LightSource {
      * @return Radiant flux in watt.
      */
     virtual vec3 getPower() const override {
-        return getIntensity() * getArea() * static_cast<float>(M_PI);
+        return getIntensity() * getArea() * glm::two_pi<float>();
```
Comment (martinfalk) [inline include/inviwo/core/datastructures/light/directionallight.h]: two pi? Why is the area weighted by pi in the first place...

File: include/inviwo/core/datastructures/light/pointlight.h [hunk 2]
```diff
@@ -27,24 +27,21 @@
...
-    PointLight() : LightSource() { setFieldOfView(static_cast<float>(2. * M_PI)); }
+    PointLight() : LightSource() { setFieldOfView(glm::two_pi<float>()); }
     virtual ~PointLight() = default;
-    virtual PointLight* clone() const override { return new PointLight(*this); }
+    virtual PointLight* clone() const override;
 
-    virtual float getArea() const override {
-        return 4.f * static_cast<float>(M_PI) * size_.x * size_.y;
-    }
+    virtual float getArea() const override { return 4.f * glm::pi<float>() * size_.x * size_.y; }
```
Comment (martinfalk) [inline include/inviwo/core/datastructures/light/pointlight.h:44]: still inline?

File: include/inviwo/core/io/serialization/deserializer.h [hunk 3]
```diff
@@ -788,35 +812,35 @@ void Deserializer::deserialize(const std::string& key, std::vector<std::unique_p
...
-                }
-            } catch (...) {
-                handleError(IVW_CONTEXT);
-            }
-        }
-        i++;
-    }
+    size_t i = 0;
+    detail::forEachChild(rootElement_, itemKey,
+                         [&](TxElement* child) {  // In the next deserialization call do net fetch
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/deserializer.h]: weird indentation. Typo in "do net fetch"

File: include/inviwo/core/io/serialization/deserializer.h [hunk 4]
```diff
@@ -904,34 +923,33 @@ void Deserializer::deserialize(const std::string& key, std::unordered_set<T>& se
...
-                deserialize(itemKey, *std::next(container.begin(), i));
-            }
-        } catch (...) {
-            handleError(IVW_CONTEXT);
-        }
-        i++;
-    }
+    size_t i = 0;
+    detail::forEachChild(rootElement_, itemKey,
+                         [&](TxElement* child) {  // In the next deserialization call do net fetch
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/deserializer.h]: weird indentation. Typo in "do net fetch"

File: include/inviwo/core/io/serialization/deserializer.h [hunk 5]
```diff
@@ -941,25 +959,24 @@ void Deserializer::deserialize(const std::string& key, std::array<T, N>& cont,
...
-                                             key);
-            }
-        } catch (...) {
-            handleError(IVW_CONTEXT);
-        }
-        i++;
-    }
+    size_t i = 0;
+    detail::forEachChild(rootElement_, itemKey,
+                         [&](TxElement* child) {  // In the next deserialization call do net fetch
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/deserializer.h]: same

File: include/inviwo/core/io/serialization/serialization.h [hunk 6]
```diff
@@ -27,13 +27,8 @@
...
  *********************************************************************************/
 
-#ifndef IVW_SERIALIZATION_H
-#define IVW_SERIALIZATION_H
+#pragma once
 
-#include <inviwo/core/io/serialization/serializebase.h>
-#include <inviwo/core/io/serialization/serializer.h>
 #include <inviwo/core/io/serialization/serializable.h>
+#include <inviwo/core/io/serialization/serializer.h>
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/serialization.h:33]: is there any reason to keep this header?

File: include/inviwo/core/ports/dataoutport.h [hunk 7]
```diff
@@ -27,16 +27,18 @@
...
 
-#include <inviwo/core/common/inviwo.h>
 #include <inviwo/core/common/inviwocoredefine.h>
 #include <inviwo/core/datastructures/datatraits.h>
 #include <inviwo/core/ports/outport.h>
 #include <inviwo/core/ports/outportiterable.h>
 #include <inviwo/core/ports/porttraits.h>
 #include <inviwo/core/util/stringconversion.h>
+#include <inviwo/core/util/document.h>
+#include <inviwo/core/util/glmvec.h>
```
Comment (martinfalk) [inline include/inviwo/core/ports/dataoutport.h]: use #include <glm/fwd.hpp> instead and glm::uvec3 for color

File: include/inviwo/core/ports/portinspectormanager.h [hunk 8]
```diff
@@ -27,15 +27,16 @@
...
 
-#ifndef IVW_PORTINSPECTORMANAGER_H
-#define IVW_PORTINSPECTORMANAGER_H
+#pragma once
 
 #include <inviwo/core/common/inviwocoredefine.h>
-#include <inviwo/core/common/inviwo.h>
 #include <inviwo/core/network/processornetworkobserver.h>
 #include <inviwo/core/io/serialization/serializable.h>
+#include <inviwo/core/util/glmvec.h>
```
Comment (martinfalk) [inline include/inviwo/core/ports/portinspectormanager.h:35]: possible to use #include <glm/fwd.hpp>?

File: include/inviwo/core/properties/invalidationlevel.h [hunk 9]
```diff
@@ -27,11 +27,11 @@
...
  *********************************************************************************/
 
-#ifndef IVW_INVALIDATIONLEVEL_H
-#define IVW_INVALIDATIONLEVEL_H
+#pragma once
 
 #include <inviwo/core/common/inviwocoredefine.h>
-#include <inviwo/core/common/inviwo.h>
+
+#include <iosfwd>
```
Comment (martinfalk) [inline include/inviwo/core/properties/invalidationlevel.h:34]: did not know that one existed...

**Review by martinfalk — DISMISSED** [review]
wow. That were a lot of files. Nice work getting rid of superfluous includes!

File: include/inviwo/core/datastructures/light/directionallight.h (hunk 1 above)
Comment (ResearchDaniel) [reply include/inviwo/core/datastructures/light/directionallight.h]: Glm::pi()
As for pi, see the pbrt book

File: include/inviwo/core/io/serialization/deserializer.h [hunk 10]
```diff
@@ -290,6 +291,8 @@ class IVW_CORE_API Deserializer : public SerializeBase, public LogFilter {
 
     void storeReferences(TxElement* node);
 
+    TxElement* retriveChild(const std::string& key);
```
Comment (ResearchDaniel) [inline include/inviwo/core/io/serialization/deserializer.h]: Typo
Retrieve

File: include/inviwo/core/datastructures/light/pointlight.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/datastructures/light/pointlight.h:44]: that for some other time...

File: include/inviwo/core/io/serialization/deserializer.h [hunk 11]
```diff
@@ -971,14 +988,13 @@ void Deserializer::deserialize(const std::string& key, std::map<K, V, C, A>& map
...
 
-    for (child = child.begin(rootElement_); child != child.end(); ++child) {
+    detail::forEachChild(rootElement_, itemKey, [&](TxElement* child) {
         // In the next deserialization call do net fetch the "child" since we are looping...
         // hence the "false" as the last arg.
-        NodeSwitch elementNodeSwitch(*this, &(*child), false);
+        NodeSwitch elementNodeSwitch(*this, child, false);
         K childkey;
-        child->GetAttribute(comparisionAttribute, &childkey, false);
+        detail::getNodeAttribute(child, comparisionAttribute, childkey);
```
Comment (ResearchDaniel) [inline include/inviwo/core/io/serialization/deserializer.h]: Typo comparison

File: include/inviwo/core/io/serialization/deserializer.h (hunk 3 above)
Comment (petersteneteg) [reply include/inviwo/core/io/serialization/deserializer.h]: clang-fomat does that...

File: include/inviwo/core/io/serialization/serialization.h (hunk 6 above)
Comment (petersteneteg) [reply include/inviwo/core/io/serialization/serialization.h:33]: to avoid breaking stuff?

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Nice! 
Found one error, same as Martin. 
Did not manage to check all changes

File: include/inviwo/core/ports/portinspectormanager.h (hunk 8 above)
Comment (petersteneteg) [reply include/inviwo/core/ports/portinspectormanager.h:35]: this not imported much does not matter so much...

**Review by ResearchDaniel — DISMISSED** [review]

**Review by martinfalk — DISMISSED** [review]

**Review by martinfalk — APPROVED** [review]

**Pull request merged**